pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -799,8 +801,29 @@ protected void splitServiceUnitOnceAndRetry(NamespaceService namespaceService,
                                                 long startTime,
                                                 CompletableFuture<Void> completionFuture) {
         CompletableFuture<List<NamespaceBundle>> updateFuture = new CompletableFuture<>();
+        Map<String, Optional<String>> bundleToDestBroker = data.splitServiceUnitToDestBroker();
+        // 1. verify the sub bundles is correct.
+        if (bundleToDestBroker.size() != 2) {
+            String msg = format("Bundle %s should be split into 2 bundles, but got %d bundles",
+                    serviceUnit, bundleToDestBroker.size());
+            updateFuture.completeExceptionally(new BrokerServiceException.ServiceUnitNotReadyException(msg));
+            return;
+        }
+        String namespace = bundle.getNamespaceObject().toString();
+
+        // 2. get the split boundary.
+        Set<Long> boundaries = new HashSet<>();
```

Comment:
This conversion looks redundant. 
Set<Long> boundaries = new HashSet<>();
...
new ArrayList<>(boundaries)

Like mentioned above, can we pass `List<Long> boundaries` in `ServiceUnitStateData`?